src/data_type/multi.value.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+module.exports = class MultiValue {
+  /**
+   * MultiValue constructor.
+   *
+   * @param {array} items
+   */
+  constructor(items) {
+    this.originalItems = items;
+    this.items = items;
+    this.filterInstances = [];
+    this.filterTypes = [];
+  }
+
+  /**
+   * Filter values by their instance type.
+   *
+   * @param {*} type
+   *
+   * @returns {MultiValue}
+   */
+  filterInstance(type) {
+    this.filterInstances.push(type);
+    return this;
+  }
+
+  /**
+   * Filter values by their constructor name.
+   *
+   * @param {String} type
+   *
+   * @returns {MultiValue}
+   */
+  filterType(type) {
+    this.filterTypes.push(type);
```

Comment:
While this adds a new _Type name_ to the list of _allowedTypes_ (or perhaps _allowedTypeNames_)